### PR TITLE
Input group scss refactoring

### PR DESCRIPTION
**Pull request by alexandreDavid**
Using a mixin for removing duplicated code.

### CODE REVIEW

**Comment by XhmikosR** [comment]
Please retarget the PR to the master branch.

**Review by XhmikosR — DISMISSED** [review]
Please retarget the PR to the master branch.

**Comment by XhmikosR** [comment]
And make sure tests pass ofc.

File: scss/mixins/_input-group.scss [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  > .form-control,
+  > .custom-select,
+  > .input-group-prepend > .input-group-text,
+  > .input-group-append > .input-group-text,
+  > .input-group-prepend > .btn,
+  > .input-group-append > .btn {
+    padding: $padding-y $padding-x;
+    @include font-size($font-size);
+    line-height: $line-height;
+    @include border-radius($badge-border-radius);
```
Comment (ysds) [inline scss/mixins/_input-group.scss]:
```suggestion
    @include border-radius($border-radius);
```

**Review by ysds — CHANGES_REQUESTED** [review]

**Comment by MartijnCuppens** [comment]
I got mixed feelings with this one. On one hand this removes a lot of duplicate code, on the other hand, this adds a new file, with only one mixin. I guess this is a step closer an ideal solution, but we're not there yet.

Thoughts @mdo @ysds?

File: scss/_input-group.scss [hunk 2]
```diff
@@ -127,46 +127,19 @@
...
-.input-group-sm > .input-group-append > .btn {
-  padding: $input-padding-y-sm $input-padding-x-sm;
-  @include font-size($input-font-size-sm);
-  line-height: $input-line-height-sm;
-  @include border-radius($input-border-radius-sm);
+.input-group-sm {
+  @include input-group-size($input-height-sm, $input-padding-y-sm, $input-padding-x-sm, $input-font-size-sm, $input-line-height-sm, $input-border-radius-sm);
 }
 
 .input-group-lg > .custom-select,
```
Comment (vsn4ik) [inline scss/_input-group.scss]:
```
.input-group-lg > .custom-select,
.input-group-sm > .custom-select {
  padding-right: $custom-select-padding-x + $custom-select-indicator-padding;
}
```
->
```
@mixin input-group-size(...) {
  ...
  > .custom-select {
    padding-right: $custom-select-padding-x + $custom-select-indicator-padding;
  }
}
```
?

**Comment by ysds** [comment]
@MartijnCuppens We already have a file for similar sizing mixin - [mixin/_pagination.scss](https://github.com/twbs/bootstrap/blob/51375abca5769682f1d27de2e29e6e9f15ec7314/scss/mixins/_pagination.scss). Maybe we should choose to add the input group sizing mixin or remove the pagination' one for consistency.

**Comment by MartijnCuppens** [comment]
Closing this to consolidate https://github.com/twbs/bootstrap/pull/29885

**Pull request closed without merging**